urwid/listbox.py

```diff
@@ -180,7 +182,7 @@ def prev_position(self, position: int) -> int:
             raise IndexError
         return position - 1
 
-    def positions(self, reverse: bool = False):
+    def positions(self, reverse: bool = False) -> Iterable[int]:
```

Comment:
positions can be anything the list walker supports as a position: this is not limited to ints.